### PR TITLE
Switch to index 20210811-3

### PR DESCRIPTION
**Pull request by charvolant**
* Include management of "synthetic" taxa, uses as placeholders for implied taxa from lists.
* Break scoring loops
* Try to avoid having misapplied names as primary taxa

### CODE REVIEW

File: ala-name-matching-builder/src/main/java/au/org/ala/names/index/LogReporter.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2022 Atlas of Living Australia
```
Comment (Rita-C) [inline ala-name-matching-builder/src/main/java/au/org/ala/names/index/LogReporter.java:2]: Would you like to update copyright on other files? e.g. 2021-> 2022

**Review by Rita-C — APPROVED** [review]
Looks good.

**Comment by charvolant** [comment]
@Rita-C Sorry Rita, some more changes for phrase names

**Review by Rita-C — APPROVED** [review]

File: ala-name-matching-builder/src/main/java/au/org/ala/names/index/TaxonConceptInstance.java [hunk 2]
```diff
@@ -979,6 +979,8 @@ private boolean validateParent(Taxonomy taxonomy) {
      */
     private TaxonConceptInstance getResolvedParent(TaxonConceptInstance original, int steps, @Nullable List<TaxonomicElement> trace, boolean exception) {
         if (steps <= 0) {
+            if (original.isForbidden() || this.isForbidden())
```
Comment (Rita-C) [inline ala-name-matching-builder/src/main/java/au/org/ala/names/index/TaxonConceptInstance.java:982]: style comment - adding `{}` even if just one statement

File: ala-name-matching-builder/src/main/java/au/org/ala/names/index/TaxonConceptInstance.java [hunk 3]
```diff
@@ -1070,8 +1080,11 @@ private TaxonConceptInstance getResolvedAccepted(TaxonConceptInstance original,
     }
 
     private TaxonConceptInstance getResolved(TaxonConceptInstance original, int steps) {
-        if (steps <= 0)
+        if (steps <= 0) {
+            if (this.isForbidden())
+                return null;
             throw new ResolutionException("Detected possible loop resolving " + original);
```
Comment (Rita-C) [inline ala-name-matching-builder/src/main/java/au/org/ala/names/index/TaxonConceptInstance.java:1086]: Does throw get executed? Already returned?

Comment (charvolant) [reply ala-name-matching-builder/src/main/java/au/org/ala/names/index/TaxonConceptInstance.java:1086]: Yes. Forbidden is a state that only a few TCIs have.

File: ala-name-matching-builder/src/main/java/au/org/ala/names/index/TaxonConceptInstance.java (hunk 2 above)
Comment (charvolant) [reply ala-name-matching-builder/src/main/java/au/org/ala/names/index/TaxonConceptInstance.java:982]: Yuk. OK fix incoming as soon as it passes its test cases.

**Review by Rita-C — APPROVED** [review]

**Pull request merged**